### PR TITLE
EET tests for EVM Opcodes

### PR DESCRIPTION
**Pull request by shemnon**
EET tests to exercise the opcodes of the EVM. The contracts are tuned
for 6KiB transaction size limit and 300K gas limit.

There are some stub classes waiting for callable contract setup.

Signed-off-by: Danno Ferrin <danno.ferrin@hedera.com>

**Description**:
Adds EET tests to cover nearly all EVM Opcodes.  (all will be covered with follow on PRs).
Tests are tuned for 6KiB transaction size limit and 300K per call gas limit.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractPerformanceSuite.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                  contractCreate(test).bytecode("bytecode"))
+              .when()
+              .then(
+                  withOpContext(
+                      (spec, opLog) -> {
+                        var contractCall = contractCall(test, "<empty>").gas(35000000);
+                        allRunFor(spec, contractCall);
+                        Assertions.assertEquals(
+                            ResponseCodeEnum.SUCCESS, contractCall.getLastReceipt().getStatus());
+                      })));
```
Comment (tinker-michaelj) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractPerformanceSuite.java:68]: I think the source spec for this `contractCall` pattern may have needed additional customization; here we could just use,

```suggestion
			hapiSpecs.add(
					defaultHapiSpec("Perf_" + test).given(
							fileCreate("bytecode").path(PERF_RESOURCES + test),
							contractCreate(test).bytecode("bytecode")
					).when( ).then(
							contractCall(test, "<empty>").gas(35000000)
					)
			);
```

**Review by tinker-michaelj — COMMENTED** [review]
Looks awesome 😁...when I run locally, two tests fail:
```
2021-09-08 15:34:41.819 INFO   178  ContractPerformanceSuite - Spec{name=Perf_returndatacopy.bin, status=FAILED}
2021-09-08 15:34:41.819 INFO   178  ContractPerformanceSuite - Spec{name=Perf_returndatasize.bin, status=FAILED}
```
...in both cases with,
```
2021-09-08 15:34:20.138 WARN   217  HapiSpecOperation - 'Perf_returndatacopy.bin' - HapiContractCall{sigs=1, node=0.0.3, contract=returndatacopy.bin, abi=<empty>, params=[]} failed Wrong status! Expected SUCCESS, was INVALID_SOLIDITY_ADDRESS
```

Do you see this?

**Comment by shemnon** [comment]
Yes, I still need to do some contract setup for a whole batch of contracts.  I need to do some search and replace on the bin since we cannot rely on contracts having the same address.  Then they should work.

File: test-clients/src/main/resource/contract/performance/performanceContracts.txt [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+swap11.bin
+swap12.bin
+swap13.bin
+swap14.bin
+swap15.bin
+swap16.bin
+timestamp.bin
+xor.bin
+
+## Contract dependant
```
Comment (shemnon) [inline test-clients/src/main/resource/contract/performance/performanceContracts.txt:141]: These are the contracts that will depend on other deployed contracts.  I'll need to do some search/replace to apply the hedera addresses.

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractPerformanceSuite.java (hunk 1 above)
Comment (shemnon) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractPerformanceSuite.java]: I think what I was missing was the `.via(name)` method. That saved the contract call in the registry by the via name and I could then get the receipt by name and check status.

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Comment by shemnon** [comment]
@tinker-michaelj  My pushes invalidated the PR approval.

**Comment by shemnon** [comment]
@tinker-michaelj I think my pushes wound up dismissing your approvals.  Can you re-approve?

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Pull request merged**